nautobot/extras/models/secrets.py

```diff
@@ -144,6 +125,8 @@ class SecretsGroupAssociation(BaseModel):
     access_type = models.CharField(max_length=32, choices=SecretsGroupAccessTypeChoices)
     secret_type = models.CharField(max_length=32, choices=SecretsGroupSecretTypeChoices)
 
+    natural_key_field_names = ["secrets_group", "access_type", "secret_type", "secret"]
```

Comment:
Added because the default natural-key derived from `unique_together` didn't include the actual `secret` as a field.

Reply:
We probably need an issue after #2908/#2909 to audit the auto-derived natural keys for all models and make sure they're as desired. Some of the auto-keys are pretty gnarly at present.

Reply:
I opened #3752 for this work.